File: transacoes.client/src/models/Transacao.ts
import { Categoria } from "./Categoria";
import { Pessoa } from "./Pessoa";

export class Transacao {
	id = 0;
	descricao = "";
	valor = 0.0;
	tipo = 0;
	idCategoria = 0;
	idPessoa = 0;
	categoria: Categoria | null = null;
	pessoa: Pessoa | null = null;
}

File: Transacoes.Server.BancoDeDados/Entidades/Transacao.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Transacoes.Server.BancoDeDados.Enumeradores;
using Transacoes.Server.BancoDeDados.Interfaces;

namespace Transacoes.Server.BancoDeDados.Entidades
{
    public class Transacao : IEntidade
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int Id { get; set; }

        [Required]
        public string Descricao { get; set; } = string.Empty;

        [Required]
        public decimal Valor { get; set; }

        [Required]
        public EnumTipoTransacao Tipo { get; set; }

        [ForeignKey("Categoria")]
        public int IdCategoria { get; set; }

        [ForeignKey("Pessoa")]
        public int IdPessoa { get; set; }

        public Categoria? Categoria { get; set; }

        public Pessoa? Pessoa { get; set; }
    }
}
